cohortextractor/main.py

```diff
@@ -14,7 +14,7 @@ def main(definition_path, output_file, backend_id, db_url, dummy_data_file=None)
     cohort = load_cohort(definition_path)
 
     output_file.parent.mkdir(parents=True, exist_ok=True)
-    if dummy_data_file:
+    if dummy_data_file and not db_url:
```

Comment:
This looks good.

One thing it might be worth doing is logging the fact that a dummy data file was supplied but it's not being used because database url was set. And by logging I think I just mean `print()` at this stage :) Anything written to stdout or stdin will end up in the job logs.